### PR TITLE
Feature/views

### PR DESCRIPTION
**Pull request by Salimer**
## This PR is raised to implement the following tasks:

- [x] Implement the design from the sneak peek wireframes.

    - Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)

*Note: Analyze the wireframes deeply and try to identify all the necessary elements and how a user would navigate from one page to another*

- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

*Hint: In order to see some posts or comments, you might need to add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).*

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
   <% @users.each do |user| %>
-    <li>id: <%= user.id %>
-    <span>name: <%= user.name %></span>
-    </li>
+    <a href="<%= user_posts_path(user) %>">
+      <div class="user-box">
+        <%= render partial: 'shared/user_photo', locals: { user: user } %>
+        <%= render partial: 'shared/user_details', locals: { user: user } %>
+      </div>
+    </a>
```
Comment (DeliceLydia) [inline app/views/users/index.html.erb:9]: - [ ] User page should render the user's 3 most recent post but currently it is showing all posts please address that

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -1,2 +1,16 @@
...
+    <span>Comments: <%= @post.comments_counter %>,</span>
+    <span>Likes: <%= @post.likes_counter %></span>
+</div>
+<p><%= @post.text %></p>
+<h3>Comments:</h3>
+<% @post.comments.each do |comment| %>
+<div class="post-comments">
+    <span><%= comment.author.name %>:</span>
+    <span><%= comment.text %></span>
+</div>
```
Comment (DeliceLydia) [inline app/views/posts/show.html.erb:15]: - [ ] A post should render 5 most recent comments but currently, it is showing 3 please address that

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @Salimer, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The Views are associated with the Models and Controllers
- Meaningful commit messages
- No linter errors


## Required Changes ♻️
_Check the comments under the review._



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Salimer** [comment]
Thank you @DeliceLydia for the thorough review, I addressed all required changes 🥳

**Review by KDlamini — APPROVED** [review]
### Hi @Salimer 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes for the Blog app views. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## Highlights 🌞

✔️  Good PR title and description
✔️  Users controller `#index` action renders all users to their corresponding Views index page
✔️  The `#recent_post` method renders the user's 3 most recent posts.
✔️  `See all Post` link opens the user's post index as displayed on the wireframe
✔️  Post show page displays post with all comments

# Status: Approved ✔️ 

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Salimer** [comment]
Thank you so much @KDlamini

**Pull request merged**